src/components/header/header.js

```diff
@@ -0,0 +1,55 @@
+import React, { useState, useEffect, useContext } from 'react';
+import Logo from '../logo';
+import Search from '../search';
+import ModalWindow from '../modalWindows/modalWindow';
+import AddMovie from '../modalWindows/addMovie';
+import MovieDetails from '../movieDetails';
+import MockDataContext from '../../context';
+import { Button } from 'react-bootstrap';
+
+import './header.css'
+
+const Header = (props) => {
+    const [show, setShow] = useState(false);
+    const [showDetails, setShowDetails] = useState(false);
+    
+    const handleClose = () => setShow(false);
+    const handleShow = () => setShow(true);
+    const handleShowDetails = () => setShowDetails(!!id);
+    const handleCloseDetails = () => setShowDetails(false);
+    
+    const movies = useContext(MockDataContext);
+
+    const { id } = props;
+
+    useEffect(() => {
+        handleShowDetails();
+      }, [id]);
```

Comment:
`handleShowDetails` is a extra function? please update state in `useEffect`

Reply:
done